### DISCUSSION TITLE
Preventing `RequestTimeTooSkewed` from happening in the browser

### DISCUSSION THREAD

**Discussion by nuthinking** [aws/aws-sdk-js#4082] [General]
Confirm by changing [ ] to [x] below:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers

**Describe the question**
If, for any reason, I change manually the time of my computer to a different one than the actual one (e.g. a day before) when I try to upload a file using this library I get the `RequestTimeTooSkewed` error. This happens even if I set `AWS.config.correctClockSkew = true` or `myS3.config.correctClockSkew = true`.

Is there any way to prevent this error from happening? Thanks!

**Comment by stagr** [comment]
Similar issue in aws-sdk-js-v3: https://github.com/aws/aws-sdk-js-v3/issues/2208

**Comment by ajredniwja** [comment]
Hey @nuthinking thanks for opening this issue, I suspect that is because the config gets overwritten with your constructor. 

The way to workaround that in V2 is mentioned here https://github.com/aws/aws-sdk-js/issues/399#issuecomment-233057244

**Comment by nuthinking** [comment]
@ajredniwja that code can fix S3 but not MediaEncoder.

**Comment by ajredniwja** [comment]
I can try to work with it, can you share repro code?

**Comment by nuthinking** [comment]
Basically, I uploaded a file on s3 (thanks to the fix above) and then created a job with `MediaConvert.createJob(params)`, the job creation fails. I get a CORS error or a Network Fail error depending on if after the upload on s3 I adjust `AWS.config.systemClockOffset`. I ended up creating a proxy for MediaEncoder.
